### PR TITLE
JIT: Refactor post phase checks

### PR DESCRIPTION
**Pull request by AndyAyersMS**
Add finer-grained notions of what post-phase checks are enabled. Start checking some IR and FG invariants earlier than we did before.

This is largely done in anticipation of moving pred list building earlier (#80193). We should also consider moving some of the IR checking earlier as well.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
@BruceForstall PTAL
cc @dotnet/jit-contrib

**Review by BruceForstall — APPROVED** [review]
Nice.

But it looks like maybe you've introduced a crash in R2R of SPC.dll?

**Comment by AndyAyersMS** [comment]
Looks like `optInit` must happen for all compiler instances, not just root method instances. Odd that SPMI replay didn't catch this one.

**Review by kunalspathak — APPROVED** [review]
LGTM

File: src/coreclr/jit/compiler.cpp [hunk 1]
```diff
@@ -1849,13 +1849,12 @@ void Compiler::compInit(ArenaAllocator*       pAlloc,
 
     fgInit();
     lvaInit();
+    optInit();
```
Comment (BruceForstall) [inline src/coreclr/jit/compiler.cpp:1852]: Seems fine, but presumably none of the optimizer phases run in the inliner-only `Compiler`, right? Was the new checking invoking checks related to optimizer data, during inlining?

Comment (AndyAyersMS) [reply src/coreclr/jit/compiler.cpp:1852]: The loop table checker is enabled after all phases and is supposed to decide when it should run. To do this It was reading `optLoopCount` which was not initialized for inlinee compilers.

Seemed like a good idea to just initialize it always.

Comment (AndyAyersMS) [reply src/coreclr/jit/compiler.cpp:1852]: While we're on the topic, I wondered if I should just make all check enablement explicit and not rely on the the checks themselves to decide when they should run enabled. Or convert the currently guarded checks to run always and internally decide when they actually checked things. So I left them both as is ....

Comment (kunalspathak) [reply src/coreclr/jit/compiler.cpp:1852]: Atleast these checks which are always on?

```
 PhaseChecks::CHECK_EH | PhaseChecks::CHECK_LOOPS | PhaseChecks::CHECK_UNIQUE | PhaseChecks::CHECK_PROFILE;
```

**Comment by AndyAyersMS** [comment]
Failure on linux arm64 checked looks like #70486. Am going to rerun.

**Pull request merged**